pkg/devfile/adapters/common/command_supervisor.go

```diff
@@ -32,9 +32,12 @@ func newSupervisorInitCommand(command common.DevfileCommand, adapter commandExec
 }
 
 // newSupervisorStopCommand creates a command implementation that stops the specified command via the supervisor
-func newSupervisorStopCommand(command common.DevfileCommand, executor commandExecutor) (command, error) {
-	cmd := []string{SupervisordBinaryPath, SupervisordCtlSubCommand, "stop", "all"}
-	if stop, err := newOverriddenSimpleCommand(command, executor, cmd); err == nil {
+func newSupervisorStopCommand(command common.DevfileCommand, cmd string, executor commandExecutor) (command, error) {
+	if cmd == "" {
+		cmd = "all"
+	}
+	cmdLine := []string{SupervisordBinaryPath, SupervisordCtlSubCommand, "stop", cmd}
```

Comment:
Why not stop all and restart?

Reply:
we do not want to stop all, if  `restart:false`, 
`odo push`  
`odo push` // do not stop run command here 
`odo push --debug` // stop run command here
`odo push`// stop debug command here

Reply:
now no need